### ISSUE TITLE
Modify X.U.ClickableWorkspaces to work with X.A.DynamicWorkspaces

### ISSUE THREAD

**Issue by oogeek** [xmonad/xmonad-contrib#510]
### Problem Description

I am wondering if it is necessary or useful to modify `X.U.ClickableWorkspaces` to work with `X.A.DynamicWorkspaces`.
For single monitor, I think the change is quite simple. I would be happy to open a PR if it is considered useful.

After some reading, the only limitation here comes from `getWsIndex`. 
```
-- | Lookup the index of a workspace id in the user's config, return Nothing
-- if that workspace does not exist in the config.
getWsIndex :: X (WorkspaceId -> Maybe Int)
getWsIndex = do
    spaces <- asks (workspaces . config)
    return $ flip elemIndex spaces
```
It only uses predefined workspaces.
I think for single monitor, a minor change would be good enough.

### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

**Comment by oogeek** [comment]
I find out that `getSortByIndex` from `X.U.WorkspaceCompare` do address the issue. However, the issue exists in `X.U.ClickableWorkspaces`

**Comment by liskin** [comment]
Can you elaborate how exactly did you solve the issue using `getSortByIndex`? I have a rough idea what the problem is but if you have a working fix for `ClickableWorkspaces` that would be marvellous.

(I don't think I'll have time to fix this myself this week.)

**Comment by oogeek** [comment]
So when the newly added workspace is passed to `getWsIndex`, it will return a `Nothing`. `getSortByIndex` manages this by using compare functions. So import a `getSortByIndex` will be helpful here.

**Comment by slotThe** [comment]
Closed via #512
